Added scope checkbox

ch1y0w0 opened the pull request:
To address the issue of websites loading scripts from external sources, I've implemented a new checkbox feature. When enabled, it filters the URLs in the URL box to only include those from the same domain as the current website. Additionally, it filters the endpoints to display only those extracted from JavaScript files within the same domain.

Commit:
Added Scope checkbox to filter results

Commit:
Updated CSS style and popup.js for some bugs

SharokhAtaie commented:
Thanks for the contribution!
As i checked, i couldn't find any use for `processStoreData()` function. Could you check it please?
Also the checkbox style doesn't match the overall extension style. If possible, fix that too.

Commit:
Updated the css to match the styles, Saved the checkbox state

ch1y0w0 commented:
Thanks for informing me. I changed to CSS style of the checkbox and its label to ensure it matches the overall extension style.
I've also corrected the usage of `processStoredData()` function and added `restoreCheckboxState()` function to save the checkbox state.

Commit:
Fixed checkbox save and lazy loading

ch1y0w0 commented:
New commit fixed checkbox state saving and lazy loading issue

Commit:
Added New CDNs to exclude list, Changed checkbox label
